### ISSUE TITLE
Bookmarks not updating

### ISSUE THREAD

**Issue by JJDLA** [nt1m/livemarks#147]
Firefox 65.0
My bookmark feeds are not updating.  I tried to force an update by changing the refresh interval from 5 minutes to 1 minute.  Six hours later there have been no new feeds.  Is there some other method to force a refresh?

**Comment by nt1m** [comment]
@JJDLA Do they appear on the settings page (we've had reports of them not appearing there) ? What feed URL(s) is affected ?

**Comment by JJDLA** [comment]
Yes, all the feeds are listed on the settings page.  The news sites are the most important.  Here’s a list of my feeds:


I have been comparing the feed refresh with 

which I have installed along with the same feed list.  As much as I prefer Livemarks it has a noticeable refresh lag.  Sometimes when I quit FF and reopen the feeds update.  Sometimes that doesn’t work.

Thanks for your attention and for providing this great extension.

**Comment by JJDLA** [comment]
Tim,
Let me know if you got the screenshots along with my last reply.  If not I’ll resend as text.
Thanks

> Begin forwarded message:
> 
> From: Alexander Daniele <alexanderdaniele@gmail.com>
> Subject: Re: [nt1m/livemarks] Bookmarks not updating (#147)
> Date: January 22, 2019 at 5:55:46 PM EST
> To: nt1m/livemarks <reply@reply.github.com>
> Cc: Alexander Daniele <alexanderdaniele@gmail.com>
> 
> Yes, all the feeds are listed on the settings page.  The news sites are the most important.  Here’s a list of my feeds:
> 
> 
> I have been comparing the feed refresh with 
> 
> which I have installed along with the same feed list.  As much as I prefer Livemarks it has a noticeable refresh lag.  Sometimes when I quit FF and reopen the feeds update.  Sometimes that doesn’t work.
> 
> Thanks for your attention and for providing this great extension.

**Comment by treba123** [comment]
Have the same issue. Context menu items are also gone, so can't refresh manually. Tried disabling and reenabling the addon, now the live bookmarks are not recognised any more by the addon.
When trying to readd them, nothing happens when clicking on `Add Livemark`. Console only prints the message: `Error: An unexpected error occurred` at line `undefined` :(

**Comment by evilpie** [comment]
That sounds a lot like `browser.sync.storage` is broken.

**Comment by Summerwood294** [comment]
I'm having the same issue

I can't access the preferences, add new bookmarks and my current bookmarks are not refreshing

**Comment by webstoney** [comment]
I have a similar issue but in this case browser-caching comes in the way. When refreshing a bookmark and Expires and/or Cache-control max-age are not met the feed isn't updated.

**Comment by evilpie** [comment]
When you open the webconsole (F12) on the livemarks preference/options page, do you see any errors?
What happens when you enter `await browser.storage.sync.get()`?

**Comment by treba123** [comment]
Just tested that and I get `Error: An unexpected error occurred`, like above.

**Comment by nt1m** [comment]
@treba123 Hi, can you please perform the following troubleshooting steps ?
- Open the webconsole (F12) on the Livemarks preference/options page
- Enter `await browser.storage.sync.get()`
- Open the browser console (Menu > Web Developer > Browser Console)
- Share the output of the browser console. 

Thanks!

**Comment by treba123** [comment]
@nt1m: the browser console showed a lot of

> Error(s) encountered during statement execution: database disk image is malformed

So I reset my browser profile, reinstalled all addons and voilà, things work again. So most probably a firefox issue. @JJDLA , @Summerwood294 , @webstoney , can you confirm this?

Sorry for not thinking about that earlier. Will dig through some firefox issues and see if there's a open bug. If I there's one, I'll post it here so other affected people can find it.

**Comment by nt1m** [comment]
@treba123 Thanks! Yes, it seems like a known issue from the Firefox side: https://bugzilla.mozilla.org/show_bug.cgi?id=1298758

**Comment by treba123** [comment]
Maybe https://bugzilla.mozilla.org/show_bug.cgi?id=1220723 is the better place to look for updates

**Comment by treba123** [comment]
@nt1m jep, thats better. Will send it to you (and trust you and google to not share/use that information....:))

**Comment by webstoney** [comment]
> So I reset my browser profile, reinstalled all addons and voilà, things work again. So most probably a firefox issue. @JJDLA , @Summerwood294 , @webstoney , can you confirm this?

I'm not sure about the others, but in my case it's definitely another problem. As i wrote, when refreshing a feed the browser-cache is not bypassed, so although there is a new feed-version the old one is reported by the cache (a forced reload ore cache-clear fixes this). ight have to do something with the feed origin but that doesn't help much.

Is it possible to force loading from the server? Would be easiest I think.

**Comment by Summerwood294** [comment]
I reset my firefox profile and I'm still having the same issue, bookmarks won't reload and I can't add anymore bookmarks
